### PR TITLE
refactor async server tests that uses Promises (close #308)

### LINKED ISSUE #308: Refactor async server tests that uses Promises
Mocha supports promises out of the box now, so you can write:
```js
it('smthg', function() {
    return somePromise()
              .then(function() {
                  // some assertions
               });
});
```
instead of:
```js
it('smthg', function(done) {
    return somePromise()
              .then(function() {
                  // some assertions
                  done();
               })
               .catch(done);
});
```

### PR DESCRIPTION
**Pull request by miherlosev**
/cc @churkin @LavrovArtem

### CODE REVIEW

**Comment by miherlosev** [comment]
Waiting for #330

File: test/server/charset-test.js [hunk 1]
```diff
@@ -257,40 +257,32 @@ describe('Content charset', function () {
             });
         }
 
-        it('Should set content charset for manifest', function (done) {
+        it('Should set content charset for manifest', function () {
             var processedManifest = manifestProcessor.processResource(manifestSrc, null, null, getProxyUrl);
             var resourceUrl       = 'http://127.0.0.1:2000/other-resource/manifest/';
 
-            Promise
+            return Promise
```
Comment (inikulin) [inline test/server/charset-test.js]: Here and after: you can put `Promise.all` in one line now

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
You need to rebase

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by churkin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**